apps/system/flash_test/flash_test.c

```diff
@@ -53,44 +53,102 @@
 /****************************************************************************
  * Included Files
  ****************************************************************************/
-
-#include <tinyara/config.h>
-#include <tinyara/progmem.h>
-#include <sys/stat.h>
-
-#include <unistd.h>
 #include <stdio.h>
+#include <string.h>
 #include <stdlib.h>
+#include <errno.h>
+
 #include <tinyara/fs/mtd.h>
 
 /****************************************************************************
- * Pre-processor Definitions
+ * Definitions
  ****************************************************************************/
+#define BUF_SIZE   1024
 
 /****************************************************************************
- * Private data
+ * Private Data
  ****************************************************************************/
 
 /****************************************************************************
- * Public Functions
+ * Private Functions
  ****************************************************************************/
+static int flash_read(unsigned long addr, int size)
+{
+	int ret;
+	int offset;
+	int buf_idx;
+	int read_size;
+	int remaining;
+	FAR uint8_t *buffer;
+	FAR struct mtd_dev_s *dev_mtd = NULL;
+
+	if (addr == 0 || size <= 0) {
+		printf("Invalid arguments, offset 0x%x, read size %d\n", addr, size);
+		return ERROR;
+	}
+
+	dev_mtd = up_flashinitialize();
+	if (!dev_mtd) {
+		printf("up_flashinitialize Failed\n");
+		return ERROR;
+	}
+
+	buffer = (uint8_t *)malloc(BUF_SIZE);
+	if (!buffer) {
+		printf("Failed to allocate buffer %d\n", BUF_SIZE);
+		return ERROR;
+	}
+
+	offset = 0;
+	remaining = size;
 
+	printf("Read 0x%x: ", addr);
+
+	while (remaining > 0) {
+		read_size = BUF_SIZE < remaining ? BUF_SIZE : remaining;
+		ret = MTD_READ(dev_mtd, addr + offset, read_size, buffer);
+		if (ret <= 0) {
+			printf("Read Failed : %d\n", ret);
```

Comment:
In this case, we'd better free the `buffer`.

Reply:
I added `free` to avoid memory leak. Thank you